closed/src/java.base/share/classes/sun/security/ec/NativeECUtil.java

```diff
@@ -158,4 +160,61 @@ static long encodeGroup(ECParameterSpec params) {
                                        n, n.length,
                                        h, h.length);
     }
+
+    /**
+     * Returns the native pointer for the provided EC public key.
+     *
+     * @param key   the EC public key
+     * @return      the native EC public key context pointer or -1 on error
+     */
+    static long getPublicKeyNativePtr(ECPublicKey key) {
+        synchronized (key) {
+          long nativePointer = encodeGroup(key.getParams());
+          if (nativePointer != -1) {
+             try {
+                byte[] x = key.getW().getAffineX().toByteArray();
+                byte[] y = key.getW().getAffineY().toByteArray();
+                int fieldType = NativeCrypto.ECField_Fp;
+                if (key.getParams().getCurve().getField() instanceof ECFieldF2m) {
+                   fieldType = NativeCrypto.ECField_F2m;
+                }
+                if (nativeCrypto.ECCreatePublicKey(nativePointer, x, x.length, y, y.length, fieldType) == -1) {
+                   nativeCrypto.ECDestroyKey(nativePointer);
+                   nativePointer = -1;
+                }
+             } finally {
+                if (nativePointer != -1) {
+                   nativeCrypto.createECKeyCleaner(key, nativePointer);
+                }
+             }
+          }
+          return nativePointer;
+       }
```

Comment:
This indentation is inconsistent (it was wonky in my suggestion, so perhaps how that happened); please fix.

Reply:
Sorry, I copied without checking. Fixed.